pymc4/inference/utils.py

```diff
@@ -29,23 +29,42 @@ def initialize_sampling_state(
     return state.as_sampling_state(), deterministic_names
 
 
-def trace_to_arviz(pm4_trace, pm4_sample_stats):
+def trace_to_arviz(
+    trace=None,
+    sample_stats=None,
+    observed_data=None,
+    prior_predictive=None,
+    posterior_predictive=None,
+):
     """
     Tensorflow to Arviz trace convertor.
 
-    Convert a PyMC4 trace as returned by sample() to an ArviZ trace object
-    that can be passed to e.g. arviz.plot_trace().
+    Creates an ArviZ's InferenceData object with inference, prediction and/or sampling data
```

Comment:
It would be nice if we made pymc4 support InferenceData independently: that is, directly supporting the schema defined there.

We'll should still use ArviZ as a way to compute diagnostics and visualizations, but the further down we push compatibility, the better, in my opinion!

Reply:
Yeah, I tough about this and the opposite alternative (implementing `az.from_pymc4`), and then decide on this PR, that is somehow an intermediate option. I do not have a clear opinion on what should be the best choice :-)